templates/leptos-ssr/content/src/server.rs

```diff
@@ -1,19 +1,53 @@
-use leptos_spin::{render_best_match_to_stream, RouteTable, server_fn::register_explicit};
-use spin_sdk::http::{IncomingRequest, ResponseOutparam};
-use spin_sdk::http_component;
+use std::sync::Arc;
 
-#[http_component]
-async fn handle_{{project-name | snake_case}}(req: IncomingRequest, resp_out: ResponseOutparam) {
-    let mut conf = leptos::get_configuration(None).await.unwrap();
-    conf.leptos_options.output_name = "{{project-name | snake_case}}".to_owned();
+use leptos::config::get_configuration;
+use leptos_wasi::{
+    handler::HandlerError,
+    prelude::{Executor, IncomingRequest, ResponseOutparam, WasiExecutor},
+};
+use wasi::exports::http::incoming_handler::Guest;
+use wasi::http::proxy::export;
 
-    // Register server functions
-    register_explicit::<crate::app::SaveCount>();
+use crate::app::{shell, App, SaveCount};
 
-    let app = crate::app::App;
+struct LeptosServer;
 
-    let mut routes = RouteTable::build(app);
-    routes.add_server_fn_prefix("/api").unwrap();
+impl Guest for LeptosServer {
+    fn handle(request: IncomingRequest, response_out: ResponseOutparam) {
+        // Initiate a single-threaded [`Future`] Executor so we can run the
+        // rendering system and take advantage of bodies streaming.
+        let executor = WasiExecutor::new(leptos_wasi::executor::Mode::Stalled);
+        if let Err(e) = Executor::init_local_custom_executor(executor.clone()) {
+            eprintln!("Got error while initializing leptos_wasi executor: {e:?}");
+            return;
+        }
+        executor.run_until(async {
+            if let Err(e) = handle_request(request, response_out).await {
+                eprintln!("Got error while handling request: {e:?}");
+            }
+        })
+    }
+}
+
+async fn handle_request(
+    request: IncomingRequest,
+    response_out: ResponseOutparam,
+) -> Result<(), HandlerError> {
+    use leptos_wasi::prelude::Handler;
 
-    render_best_match_to_stream(req, resp_out, &routes, app, &conf.leptos_options).await
+    let mut conf = get_configuration(None).unwrap();
+    conf.leptos_options.output_name = Arc::from("{{project-name | snake_case}}".to_owned());
```

Comment:
This is kinda strange, cargo-leptos should be setting this as an env var, and get_configuration(None) should load it you shouldn't be needing to hardcode it

Reply:
@benwis This might be historical (it's existing code which GitHub is getting confused and showing as a diff).  During early integration work I found myself needing to force the project name, and I tried a bunch of ways to do that: it's quite likely more of them got left in than are strictly needed...!

Reply:
Good callout, a test Leptos Spin app does seem to function fine without this -- I went ahead and removed here and in README.

Thanks for the context @itowlson -- do you remember why the project name had to be forced?

Reply:
@ogghead I think it was to do with finding assets under the `/pkg` route - it was looking for assets using the wrong name.  But yeah this was very much an "at the time" thing.  I am happy to write it off as an accident of history that is no longer needed.  Thanks for testing and thanks @benwis for querying it!